test/threadsafe_function/threadsafe_function.js

```diff
@@ -25,11 +25,9 @@ async function test(binding) {
       binding.threadsafe_function[threadStarter](function testCallback(value) {
         array.push(value);
         if (array.length === quitAfter) {
-          setImmediate(() => {
```

Comment:
I didn't know the exact intention of deferring the close of TSFN here. However, tests all passed with or without deferring.

Reply:
@KevinEady @gabrielschulhof any input on using/not using process.nextTick().  I'll plan to land this tomorrow unless I hear from you so that we can get back to green CI.